File: cdk-from-cfn-testing/expected/ec2/typescript/Ec2Construct.ts
import * as cdk from 'aws-cdk-lib';
import * as ec2 from 'aws-cdk-lib/aws-ec2';
import { Construct } from 'constructs';

export interface Ec2ConstructProps {
}

export class Ec2Construct extends Construct {
  public constructor(scope: Construct, id: string, props: Ec2ConstructProps = {}) {
    super(scope, id);

    // Resources
    const testVpc = new ec2.CfnVPC(this, 'TestVPC', {
      cidrBlock: '10.0.0.0/16',
    });

    const sg1 = new ec2.CfnSecurityGroup(this, 'SG1', {
      groupDescription: 'SG2',
      vpcId: testVpc.ref,
      securityGroupEgress: [
        {
          ipProtocol: 'TCP',
          fromPort: 10000,
          toPort: 10000,
          cidrIp: '10.0.0.0/16',
        },
      ],
    });
  }
}

File: cdk-from-cfn-testing/expected/ec2/csharp/Ec2Construct.cs
using Amazon.CDK;
using Amazon.CDK.AWS.EC2;
using Constructs;
using System.Collections.Generic;

namespace Ec2Construct
{
    public class Ec2ConstructProps
    {
    }

    public class Ec2Construct : Construct
    {
        public Ec2Construct(Construct scope, string id, Ec2ConstructProps props = null) : base(scope, id)
        {

            // Resources
            var testVpc = new CfnVPC(this, "TestVPC", new CfnVPCProps
            {
                CidrBlock = "10.0.0.0/16",
            });
            var sg1 = new CfnSecurityGroup(this, "SG1", new CfnSecurityGroupProps
            {
                GroupDescription = "SG2",
                VpcId = testVpc.Ref,
                SecurityGroupEgress = new []
                {
                    new CfnSecurityGroup.EgressProperty
                    {
                        IpProtocol = "TCP",
                        FromPort = 10000,
                        ToPort = 10000,
                        CidrIp = "10.0.0.0/16",
                    },
                },
            });
        }
    }
}
